[stable/datadog] Add labels recommended by Helm

rifelpet opened the pull request:
#### What this PR does / why we need it:

Updates the resources in datadog to use standard label keys recommended by helm.

See https://helm.sh/docs/chart_best_practices/#standard-labels

#### Which issue this PR fixes
  - N/A

#### Special notes for your reviewer:

Since the DaemonSet and Deployment selectors are immutable, this is breaking change and will require a deletion and recreation, hence the major version bump.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

clamoriniere commented on stable/datadog/templates/agent-apiservice.yaml:
```diff
@@ -4,10 +4,10 @@ kind: APIService
 metadata:
   name: v1beta1.external.metrics.k8s.io
   labels:
-    app: "{{ template "datadog.fullname" . }}"
```
As a first migration step, could we have both labels (old and new recommended labels) ?

clamoriniere reviewed (commented):
Hi @rifelpet 
Thank for your contribution.

We wanted to propose to our chart user a smoother migration approach to the new "recommended helm tracking labels". And so avoid to bump the "major" digit in our chart version.

What do you think about doing the migration in 2 steps:
* Add new labels without removing the olds, in a new minor chart version bump.
* Remove old labels, in a major chart version bump.
Like this, users that monitor the chart deployment using old labels, can migrate easily to the new labels.

rifelpet committed:
[stable/datadog] Add labels recommended by Helm

See https://helm.sh/docs/chart_best_practices/#standard-labels

New labels are added, no existing labels are removed nor any label selectors updated.

Signed-off-by: Peter Rifel <pgrifel@gmail.com>
